### PR TITLE
Refactor clientprofile mutations (DEV-678, DEV-702)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-702
DEV-678

### CODE REVIEW

File: apps/betterangels-backend/accounts/schema.py [hunk 1]
```diff
@@ -181,39 +208,6 @@ def create_client_profile(self, info: Info, data: CreateClientProfileInput) -> C
                 },
             )
 
-            if contacts_data:
```
Comment (mikefeldberg) [inline apps/betterangels-backend/accounts/schema.py:184]: turns out we don't need this

Comment (mikefeldberg) [inline apps/betterangels-backend/accounts/schema.py:184]: turns out strawberry is smart enough to create the related objects. I don't remember exactly when/why we adopted this pattern (maybe when we had related objects with custom logic and couldn't use the resolver directly) but it's not necessary for create.

Comment (lena) [reply apps/betterangels-backend/accounts/schema.py:184]: 😮

**Review by lena — APPROVED** [review]
nice refactor 🎠

File: apps/betterangels-backend/accounts/schema.py [hunk 2]
```diff
@@ -44,6 +39,43 @@
...
+    model_class_string: str,
+    data: List[Dict[str, Any]],
+    client_profile: ClientProfile,
+) -> None:
+    model_class = apps.get_model("accounts", model_class_string)
+
+    item_updates_by_id = {item["id"]: item for item in data if item.get("id")}
+    items_to_create = [item for item in data if not item.get("id")]
+    items_to_update = model_class.objects.filter(id__in=item_updates_by_id.keys(), client_profile=client_profile)
+    model_class.objects.exclude(id__in=item_updates_by_id).delete()
```
[the remark this answers is not in the document]
Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/schema.py]: good call

File: apps/betterangels-backend/accounts/schema.py [hunk 3]
```diff
@@ -164,14 +196,9 @@ def create_client_profile(self, info: Info, data: CreateClientProfileInput) -> C
         with transaction.atomic():
```
[the remark this answers is not in the document]
Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/schema.py:200]: we'll have nicer errors here later

**Pull request merged**